lib/src/chart/line_chart/line_chart.dart

```diff
@@ -82,31 +85,46 @@ class _LineChartState extends AnimatedWidgetBaseState<LineChart> {
           touchData.touchCallback(response);
         }
       },
-      onPanCancel: () {
+      onTapCancel: () {
         final Size chartSize = _getChartSize();
         if (chartSize == null) {
           return;
         }
 
-        final LineTouchResponse response = _touchHandler?.handleTouch(
-            FlPanEnd(Offset.zero, Velocity(pixelsPerSecond: Offset.zero)), chartSize);
+        final LineTouchResponse response =
+            _touchHandler?.handleTouch(FlTap(_lastTouchedPosition), chartSize);
         if (_canHandleTouch(response, touchData)) {
           touchData.touchCallback(response);
         }
       },
-      onPanEnd: (DragEndDetails details) {
+      onTapDown: (details) {
         final Size chartSize = _getChartSize();
         if (chartSize == null) {
           return;
         }
 
         final LineTouchResponse response =
-            _touchHandler?.handleTouch(FlPanEnd(Offset.zero, details.velocity), chartSize);
+            _touchHandler?.handleTouch(FlPanStart(details.localPosition), chartSize);
         if (_canHandleTouch(response, touchData)) {
+          setState(() {
+            _lastTouchedPosition = details.localPosition;
+          });
           touchData.touchCallback(response);
         }
       },
-      onPanDown: (DragDownDetails details) {
+      onTapUp: (details) {
+        final Size chartSize = _getChartSize();
+        if (chartSize == null) {
+          return;
+        }
+
+        final LineTouchResponse response =
+            _touchHandler?.handleTouch(FlTap(details.localPosition), chartSize);
+        if (_canHandleTouch(response, touchData)) {
+          touchData.touchCallback(response);
+        }
+      },
+      onHorizontalDragStart: (DragStartDetails details) {
```

Comment:
Do you think we just need horizontal?
do we need vertical direction too?
I think it's needed when we have a vertical scroll view.

Reply:
I don't see how we could need a vertical gesture because we only have a vertical touchline, which respond to a horizontal gesture. If we had a horizontal touchline, it would be a different story. But linecharts are left to right, never up and down, so I don't see how we would ever need it.

Reply:
We have to try a solution that works on all type of charts
and even in the LineChart, maybe you have some spots in the same x-axis and different y values.

Reply:
The long press still exists for those points. A vertical drag would, in theory, only be needed if he had a horizontal touchline to start swiping up and down.

Reply:
See my comment below if that's acceptable

Reply:
Also: this solution is only for the line chart. The other charts use separate gestures in separate files. 

Regarding multiple different y values. Even if a user has multiple different values, he doesn't want his chart to stop his scrollview from scrolling. That's why we block this gesture, it conflicts with standard mobile UX behavior. They still have access to Y points up and down by starting with the correct gesture long press, tap or right swipe. Just not upswipe.